### DISCUSSION TITLE
FireAngel Module ZigBee does not pair

### DISCUSSION THREAD

**Discussion by amoebapl** [Koenkk/zigbee2mqtt#6137] [General]
Hey.
I recently bought a FireAngel NM-CO-10X carbon monoxide detector + ZigBee module.
I wanted to add it to my ZigBee (HA) network. It is on the list of supported devices.
I tried on CC2531, CC2652RB (I have both) and nothing, this device has not been added.
My other XIAOMI devices connect and work without problem.
CC2652RB has the latest firmware uploaded - CC2652RB_coordinator_20210120.hex, Z2M v 1.17.1
I used a sniffer and I can see the module sending packets when trying to pair
Below is data from Wireshark when pairing this module:

1 0.000 0x0000 ZigBee transmission 58 Link status
2 0.249 Broadcast IEEE 802.15.4 21 Beacon Request
3 0.251 0x0000 ZigBee 39 Beacon, Src: 0x0000, EPID: dd: dd: dd: dd: dd: dd: dd: dd
4 9.352 Broadcast IEEE 802.15.4 21 Beacon Request
5 9.354 0x0000 ZigBee 39 Beacon, Src: 0x0000, EPID: dd: dd: dd: dd: dd: dd: dd: dd
6 15.146 0x0000 Broadcast ZigBee 58 Link status
7 19.4580 Broadcast IEEE 802.15.4 21 Beacon Request
8 19.4597 0x0000 ZigBee 39 Beacon, Src: 0x0000, EPID: dd: dd: dd: dd: dd: dd: dd: dd
And so on.

Nothing shows up in the Zigbee2mqtt logs during pairing.

Any ideas?

PS.
I found the specification of this module.
Maybe it will be useful for those more experienced in this topic and will help to solve the pairing problem.
https://www.eps-vertrieb.de/downloads/file/id/fd9a9593f3a3f1720b4654ea895a470b

**Comment by bderuyck** [comment]
I have both a smoke detector and a co detector from fire angel. Both with a ZB-module. 

At first I could pair both , but after a while the connection is lost. I can't pair  any of them anymore. 
I have put in new ZB-modules, but nothing works. 

I'm using the Sonoff Zigbee 3.0 coordinator and put the devices close to it ( < 30 cm ).

**Comment by Jbugger** [comment]
Same issue here. My smoke detector FireAngel AngelEye STAE630 with Zigbee module does seem to send a Beacon Request, and my dongle connected to Zigbee2MQTT does nicely send a Beacon Response back. Then, it seems the smoke detector either does not detect the incoming Beacon Response, or does not take further action upon receiving it. Maybe it expects some bits to be set in the Beacon Response?

**Comment by Jbugger** [comment]
I made a packet capture. The smoke detector FireAngel AngelEye ST-AE630 w/ Zigbee module sends the Beacon Request first:

Beacon Request packet bytes:
```
0000   03 08 b3 ff ff ff ff 07
```
Beacon Request packet:
```
Frame 10: 8 bytes on wire (64 bits), 8 bytes captured (64 bits) on interface COM10, id 0
IEEE 802.15.4 Command, Dst: Broadcast
    Frame Control Field: 0x0803, Frame Type: Command, Destination Addressing Mode: Short/16-bit, Frame Version: IEEE Std 802.15.4-2003, Source Addressing Mode: None
        .... .... .... .011 = Frame Type: Command (0x3)
        .... .... .... 0... = Security Enabled: False
        .... .... ...0 .... = Frame Pending: False
        .... .... ..0. .... = Acknowledge Request: False
        .... .... .0.. .... = PAN ID Compression: False
        .... .... 0... .... = Reserved: False
        .... ...0 .... .... = Sequence Number Suppression: False
        .... ..0. .... .... = Information Elements Present: False
        .... 10.. .... .... = Destination Addressing Mode: Short/16-bit (0x2)
        ..00 .... .... .... = Frame Version: IEEE Std 802.15.4-2003 (0)
        00.. .... .... .... = Source Addressing Mode: None (0x0)
    Sequence Number: 179
    Destination PAN: 0xffff
    Destination: 0xffff
    Command Identifier: Beacon Request (0x07)
```
Response from my CC2531 USB stick with firmware CC2531_20190425.zip (from https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.0.x/bin/CC2531_20190425.zip)

Beacon Response packet bytes:
```
0000   00 80 71 62 1a 00 00 ff cf 00 00 00 22 84 5f 28
0010   da 14 00 4b 12 00 ff ff ff 00
```
Beacon Response packet :
```
Frame 11: 26 bytes on wire (208 bits), 26 bytes captured (208 bits) on interface COM10, id 0
IEEE 802.15.4 Beacon, Src: 0x0000
    Frame Control Field: 0x8000, Frame Type: Beacon, Destination Addressing Mode: None, Frame Version: IEEE Std 802.15.4-2003, Source Addressing Mode: Short/16-bit
        .... .... .... .000 = Frame Type: Beacon (0x0)
        .... .... .... 0... = Security Enabled: False
        .... .... ...0 .... = Frame Pending: False
        .... .... ..0. .... = Acknowledge Request: False
        .... .... .0.. .... = PAN ID Compression: False
        .... .... 0... .... = Reserved: False
        .... ...0 .... .... = Sequence Number Suppression: False
        .... ..0. .... .... = Information Elements Present: False
        .... 00.. .... .... = Destination Addressing Mode: None (0x0)
        ..00 .... .... .... = Frame Version: IEEE Std 802.15.4-2003 (0)
        10.. .... .... .... = Source Addressing Mode: Short/16-bit (0x2)
    Sequence Number: 113
    Source PAN: 0x1a62
    Source: 0x0000
    Superframe Specification: PAN Coordinator, Association Permit
        .... .... .... 1111 = Beacon Interval: 15
        .... .... 1111 .... = Superframe Interval: 15
        .... 1111 .... .... = Final CAP Slot: 15
        ...0 .... .... .... = Battery Extension: False
        .1.. .... .... .... = PAN Coordinator: True
        1... .... .... .... = Association Permit: True
    GTS
        GTS Descriptor Count: 0
        GTS Permit: False
    Pending Addresses: 0 Short and 0 Long
ZigBee Beacon, ZigBee PRO, EPID: TexasIns_00:14:da:28:5f
    Protocol ID: 0
    Beacon: Stack Profile: ZigBee PRO, Router Capacity, End Device Capacity
        .... .... .... 0010 = Stack Profile: ZigBee PRO (0x2)
        .... .... 0010 .... = Protocol Version: 2
        .... .1.. .... .... = Router Capacity: True
        .000 0... .... .... = Device Depth: 0
        1... .... .... .... = End Device Capacity: True
    Extended PAN ID: TexasIns_00:14:da:28:5f (00:12:4b:00:14:da:28:5f)
    Tx Offset: 16777215
    Update ID: 0
```
After this response the smoke detector tries this sequence again.
I'm still curious what the smoke detector needs more to start the association. Does it maybe need a higher protocol version (currently 2)?
